.github/workflows/cibuildwheel.yml

```diff
@@ -0,0 +1,142 @@
+name: cibuildwheel
+
+# Note: We use a dynamic matrix to build different sets of wheels under
+# different conditions. On workflow_dispatch, we build the full suite of
+# wheels. This takes hours, so on pull_request, we just build a representative
+# sample.
+
+# The full list of cibuildwheel's build targets can be found here:
+# https://github.com/pypa/cibuildwheel/blob/v2.2.0a1/cibuildwheel/resources/build-platforms.toml
+
+# Notes on build targets we (don't) support:
+# - pypy: libtorrent doesn't build with pypy as of writing
+# - macos_arm64: can be cross-compiled from x86_64, but not run, so can't be
+#   tested. Build output indicates it isn't building correctly
+# - macos_universal2: b2 / setup.py doesn't have a straightforward way to build
+#   this as of writing
+# - abi3: Not supported by boost-python (or pybind11) or cibuildwheel as of
+#   writing
+
+on:
+  workflow_dispatch:
+    inputs:
+      publish:
+        description: Write 'PUBLISH' to publish to pypi. BEWARE! ARTIFACTS ARE IMMUTABLE AND CANNOT BE REPLACED ONCE PUBLISHED!
+      publish_test:
+        description: Write 'PUBLISH_TEST' to publish to test-pypi. BEWARE! ARTIFACTS ARE IMMUTABLE AND CANNOT BE REPLACED ONCE PUBLISHED!
+
+  pull_request:
+    paths:
+      - .github/workflows/cibuildwheel.yml
+      - tools/cibuildwheel/**
+      - pyproject.toml
```

Comment:
this is assuming that anything that would break cibuildwheel would also break the existing builds of the python bindings, right?
Isn't this the only place where the binding is built in an anylinux environment though?

Reply:
Changes to the referenced files could break `cibuildwheel` without breaking builds of the python bindings.

`cibuildwheel` is the easiest (or only) way to use `manylinux`/`musllinux` environments, yes.

Not sure if this answers what you're asking, but I expect `cibuildwheel` to be redundant with the other python build workflows, except when we want the specialized environments for building release wheels.

Reply:
In theory we could remove the existing python bindings workflows and just use `cibuildwheel` with limited flavor selection instead for all PRs, but I think there are a few disadvantages:
* The `cibuildwheel` workflow builds openssl from source on Linux, which takes longer
* The `cibuildwheel` workflow wouldn't test against distro versions of openssl, which can be nice